File: app/src/main/java/net/mmho/photomap2/AddressListAdapter.java
package net.mmho.photomap2;

import android.content.Context;
import android.location.Address;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

public class AddressListAdapter extends ArrayAdapter<Address>{

    private int resource;
    private LayoutInflater inflater;

    public AddressListAdapter(Context context, int resource,Address[] addresses) {
        super(context,resource,addresses);
        this.resource = resource;
        inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        View v;
        if(convertView!=null){
            v = convertView;
        }
        else{
            v = inflater.inflate(resource,null);
        }
        TextView text1 = (TextView) v.findViewById(android.R.id.text1);
        TextView text2 = (TextView) v.findViewById(android.R.id.text2);
        Address address = getItem(position);
        text1.setText(AddressUtil.getDescription(address));
        text2.setText(String.format("%6.4f,%6.4f", address.getLatitude(), address.getLongitude()));
        return v;
    }


}

File: app/src/main/java/net/mmho/photomap2/AddressListAdapter.kt
package net.mmho.photomap2

import android.content.Context
import android.location.Address
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView

import java.util.Locale

internal class AddressListAdapter(context: Context, private val resource: Int, addresses: Array<Address>)
    : ArrayAdapter<Address>(context, resource, addresses) {
    private val inflater: LayoutInflater

    init {
        inflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View? {
        val v: View = convertView ?: inflater.inflate(resource, null)
        val text1 = v.findViewById(android.R.id.text1) as TextView
        val text2 = v.findViewById(android.R.id.text2) as TextView
        val address = getItem(position)
        text1.text = AddressUtil.getDescription(address)
        text2.text = String.format(Locale.getDefault(), "%6.4f,%6.4f", address.latitude, address.longitude)
        return v
    }


}
